pkgs/development/python-modules/blivet/default.nix

```diff
@@ -0,0 +1,109 @@
+{ lib
+, pkgs
+, python
+, buildPythonPackage
+, fetchFromGitHub
+, pygobject3
+, libblockdev
+, bytesize
+, pyudev
+, dbus-python
+, util-linux
+, kmod
+, libndctl
+, nvme-cli
+, dosfstools
+, e2fsprogs
+, hfsprogs
+, xfsprogs
+, f2fs-tools
+, ntfs3g
+, btrfs-progs
+, reiserfsprogs
+, mdadm
+, lvm2
+, gfs2-utils
+, cryptsetup
+, multipath-tools
+, dracut
+, stratisd
+}:
+
+let
+  libblockdevPython = (libblockdev.override {
+    python3 = python;
+  }).python;
+in
+buildPythonPackage rec {
+  pname = "blivet";
+  version = "3.8.2";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "storaged-project";
+    repo = pname;
+    rev = "${pname}-${version}";
+    hash = "sha256-+VYhdLsgvoXt408zzgMsoavYhD/qXBsigT5EFRNsM7s=";
+  };
+
+  postPatch = ''
+    find blivet -name '*.py' | while IFS= read -r i ; do
+      substituteInPlace "$i" \
+        --replace \
+          'gi.require_version("BlockDev",' \
+          'import gi.repository
+gi.require_version("GIRepository", "2.0")
+from gi.repository import GIRepository
+GIRepository.Repository.prepend_search_path("${libblockdev}/lib/girepository-1.0")
+gi.require_version("BlockDev",'
+    done
+  '';
+
+  propagatedBuildInputs = [
+    pygobject3
+    libblockdevPython
+    bytesize
+    pyudev
+    dbus-python
+    util-linux
+    kmod
+    libndctl
+    nvme-cli
+    pkgs.systemd
```

Comment:
Propagating non-Python dependencies will not make them used.

Reply:
It's needed for `udevadm`. Don't we have to propagate it in `blivet` so that it will be in `blivet-gui`'s `PATH`?

Reply:
Oh, right, I blissfully forgot about `buildPythonApplication` adding it to wrapper implicitly.